### ISSUE TITLE
Split RecordTagHelper into a gem and remove from Rails

### ISSUE THREAD

**Issue by dhh** [rails/rails#18337]
I remember adding RecordTagHelper as an experiment, but never felt happy with its use in real life. Let's kick it out in a gem and remove it from core.

**Comment by carlosantoniodasilva** [comment]
David, I've seen `content_tag_for` widely used by many apps, and it has a simple implementation that has close to 0 cost of maintenance. I think there is no need to remove it from core. We could, however, deprecate `div_for` in my opinion, as it does not add much value on top of the former.

**Comment by dhh** [comment]
I'm not a fan of it. It seems anemic to me over just doing something like `content_tag :div, id: dom_id(record) do/end`.

**Comment by carlosantoniodasilva** [comment]
Well, technically it does add both `id` and `class` using `dom_id` and `dom_class` respectively. In any case, it is also useful for collections:

```ruby
content_tag_for :div, @posts do/end

# which would be something like:

@posts.each do |post|
  content_tag :div, id: dom_id(post), class: dom_class(post) do/end
end
```

If you still feel skeptical about having it in core then I think I have no more arguments on this, other than extracting to a gem in case people want it back while upgrading :).

**Comment by dhh** [comment]
I don’t like the look of that as it is. I remain :+1: on going gem on it.

**Comment by carlosantoniodasilva** [comment]
David, do you think it should initially live under the rails org or the author could manage the extracted gem under its user (and we would document it on release notes and so on, of course)?

**Comment by dhh** [comment]
I think traditionally, we’ve had the gem live under rails. That seems fine to continue.

**Comment by carlosantoniodasilva** [comment]
:+1:

**Comment by rafaelfranca** [comment]
:+1: for rails organization. If author want to manage the release and the repository it is fine, we can give him the access, as long we can do also. BTW, the copyright should still be the same as Rails since it an extraction of code.

**Comment by todd** [comment]
I'll commit to maintaining the gem moving forward - just let me know how you guys would prefer to handle that. As far as copyrights and things go, I can take care of the LICENSE file as that was just auto-generated by GitHub, but I'd also like some feedback on author attribution in [the gemspec](https://github.com/todd/record_tag_helper/blob/master/record_tag_helper.gemspec) - I just put all my own information in there for now as placeholders. Alternatively, if someone would like to take care of those updates themselves, I'd be more than happy to add them as a contributor on the temporary repo.

**Comment by todd** [comment]
@rafaelfranca Any more thoughts on author attribution?

**Comment by todd** [comment]
Hey all - just wanted to loop back on this. Let me know how I should proceed.

**Comment by rafaelfranca** [comment]
@todd of course the author attribution is fine. The capyright too, you just need to keep the original author and copyright attribution.

**Comment by todd** [comment]
@rafaelfranca Original author being DHH or someone else? To clarify, I only need to change that in LICENSE?

**Comment by rafaelfranca** [comment]
Original author is David. And gemspec and licence needs to be changed

**Comment by todd** [comment]
@rafaelfranca Ok, I took care of both of those files.

**Comment by rafaelfranca** [comment]
Closed by #18411
